### ISSUE TITLE
Writing movie one frame at a time

### ISSUE THREAD

**Issue by jakirkham** [Zulko/moviepy#619]
This has a bit of a Stack Overflow flavor. So if it needs to move, please let me know.

Regularly I have data that is simply too large to fit all of in memory at a time. However I can reasonably load one or perhaps a couple frames in memory at a time. I would like to write this data to say an mp4. Is there a good way to accomplish this with moviepy? I've seen some examples where all of the data must be in memory, but it would be nice to have some pointers on how to solve this when all of the data cannot be in memory at the same time. Thanks.

**Comment by jakirkham** [comment]
Hmm...I think I have answered my own question with [this]( https://github.com/Zulko/moviepy/blob/v0.2.3.2/moviepy/video/io/ffmpeg_writer.py#L130 ). Is there an example in the docs though?

**Comment by Zulko** [comment]
Actually most examples around are for cases where frames are generated "on the fly" while writing the video, using ``VideoFileClip(make_frame=some_function_that generates_frames)``. See for instance:

http://zulko.github.io/blog/2014/11/29/data-animations-with-python-and-moviepy/

**Comment by jakirkham** [comment]
Good point. Certainly something like this can be adapted very easily. Thanks.

Feel free to close.
